### PR TITLE
ddl: expose getPrimaryKey() as a public method of model.TableInfo

### LINKED ISSUE #35511: ddl refactor: wanna set getPrimaryKey() public for TiCDC to resue it
## Enhancement
The function getPrimaryKey() in `tidb/ddl` is very similar with one function in TiCDC
https://github.com/pingcap/tidb/blob/4ae78cdead54097679407314ce33b895e3c01d16/ddl/ddl_api.go#L1638-L1644

related issue in TiFlow: https://github.com/pingcap/tiflow/issues/5902

### PR DESCRIPTION
**Pull request by crelax**
### What problem does this PR solve?

Issue Number: close #35511

Problem Summary:

### What is changed and how it works?
Set the visibility of function getPrimaryKey() public 
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1628,20 +1628,20 @@ func checkInvisibleIndexOnPK(tblInfo *model.TableInfo) error {
...
 
-// getPrimaryKey extract the primary key in a table and return `IndexInfo`
+// GetPrimaryKey extract the primary key in a table and return `IndexInfo`
 // The returned primary key could be explicit or implicit.
 // If there is no explicit primary key in table,
 // the first UNIQUE INDEX on NOT NULL columns will be the implicit primary key.
 // For more information about implicit primary key, see
 // https://dev.mysql.com/doc/refman/8.0/en/invisible-indexes.html
-func getPrimaryKey(tblInfo *model.TableInfo) *model.IndexInfo {
+func GetPrimaryKey(tblInfo *model.TableInfo) *model.IndexInfo {
```
Comment (hawkingrei) [inline ddl/ddl_api.go:1644]: Why does ```GetPrimaryKey``` become public ？

Comment (crelax) [reply ddl/ddl_api.go:1644]: Hi, wanna reuse it in TiFlow (cdc). Will it cause problems?  ref: https://github.com/pingcap/tiflow/issues/5902

**Review by xiongjiwei — APPROVED** [review]

File: ddl/ddl_api.go (hunk 1 above)
Comment (hawkingrei) [reply ddl/ddl_api.go:1644]: OK, I think ```GetPrimaryKey``` can become the number function of ```model.TableInfo```.

Comment (crelax) [reply ddl/ddl_api.go:1644]: make sense

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**